inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp

```diff
@@ -738,714 +738,366 @@ struct typed_resizeLinearU8C3C4 {
+        alpha = convert_cast<alpha_type>(inv_ratio);
     }
 };
 
-GAPI_FLUID_KERNEL(FSplit2, Split2, false) {
-    static const int LPI = 4;
-    static const int Window = 1;
-    static void run(const cv::gapi::fluid::View & in,
-                    cv::gapi::fluid::Buffer & out1,
-                    cv::gapi::fluid::Buffer & out2) {
-        GAPI_DbgAssert(2 == in.meta().chan);
-        GAPI_DbgAssert(1 == out1.meta().chan);
-        GAPI_DbgAssert(1 == out2.meta().chan);
-        GAPI_DbgAssert(in.meta().depth == out1.meta().depth);
-        GAPI_DbgAssert(in.meta().depth == out2.meta().depth);
-        GAPI_DbgAssert(is_cv_type_in_list<split_supported_types>(in.meta().depth));
+namespace areaDownscale32f {
+struct Mapper: public AreaDownMapper<float, int, float> {
+    Mapper(int _inSz, int _outSz) {
+        init(_inSz, _outSz);
+    }
+};
+}
 
-        const auto rowFunc = type_dispatch<split_supported_types>(in.meta().depth, cv_type_id{}, typed_split_row<isa_tag_t, 2>{}, nullptr);
-        for (int i = 0, lpi = out1.lpi(); i < lpi; i++) {
-            std::array<uint8_t*, 2> outs = { out1.OutLineB(i), out2.OutLineB(i) };
-            rowFunc(in.InLineB(i), outs, in.length());
-        }
+namespace areaDownscale8u {
+struct Mapper: public AreaDownMapper<Q0_16, short, Q8_8> {
+    Mapper(int _inSz, int _outSz) {
+        init(_inSz, _outSz);
     }
 };
+}
 
-GAPI_FLUID_KERNEL(FSplit3, Split3, false) {
-    static const int LPI = 4;
-    static const int Window = 1;
-    static void run(const cv::gapi::fluid::View & in,
-                    cv::gapi::fluid::Buffer & out1,
-                    cv::gapi::fluid::Buffer & out2,
-                    cv::gapi::fluid::Buffer & out3) {
-        GAPI_DbgAssert(3 == in.meta().chan);
-        GAPI_DbgAssert(1 == out1.meta().chan);
-        GAPI_DbgAssert(1 == out2.meta().chan);
-        GAPI_DbgAssert(1 == out3.meta().chan);
-        GAPI_DbgAssert(in.meta().depth == out1.meta().depth);
-        GAPI_DbgAssert(in.meta().depth == out2.meta().depth);
-        GAPI_DbgAssert(in.meta().depth == out3.meta().depth);
+namespace areaUpscale {
+struct Mapper {
+    typedef short alpha_type;
+    typedef short index_type;
+    constexpr static const int unity = ONE;
 
-        GAPI_DbgAssert(is_cv_type_in_list<split_supported_types>(in.meta().depth));
+    typedef MapperUnit<short, short> Unit;
 
-        const auto rowFunc = type_dispatch<split_supported_types>(in.meta().depth, cv_type_id{}, typed_split_row<isa_tag_t, 3>{}, nullptr);
-        for (int i = 0, lpi = out1.lpi(); i < lpi; i++) {
-            std::array<uint8_t*, 3> outs = { out1.OutLineB(i), out2.OutLineB(i),
-                                            out3.OutLineB(i) };
-            rowFunc(in.InLineB(i), outs, in.length());
-        }
-    }
-};
+    static inline Unit map(double ratio, int start, int max, int outCoord) {
+        int s = cvFloor(outCoord*ratio);
+        float f = static_cast<float>((outCoord+1) - (s+1)/ratio);
```

Comment:
And `single letter variable`. It is difficult for readability.

Reply:
Done.